kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogReplayMetricsSuite.scala

```diff
@@ -294,6 +305,32 @@ class LogReplayMetricsSuite extends QueryTest
         expParquetReadSetSizes = Seq(15, 15))
     }
   }
+
+  Seq(true, false).foreach { deleteLastCheckpointMetadataFile =>
+    test("ensure `_last_checkpoint` is tried to read only once when " +
+      s"""${if (deleteLastCheckpointMetadataFile) "not exists" else "valid file exists"}""") {
+      withTempDirAndEngine { (dir, tc) =>
+        val path = dir.getAbsolutePath
+
+        for (_ <- 0 to 14) { appendCommit(path) }
+
+        if (deleteLastCheckpointMetadataFile) {
+          assert(Files.deleteIfExists(new File(path, "_delta_log/_last_checkpoint").toPath))
+        }
+
+        // there should be one checkpoint file at version 10
+        loadScanFilesCheckMetrics(
+          tc,
+          Table.forPath(tc, path),
+          expJsonVersionsRead = 14L to 11L by -1L,
+          expParquetVersionsRead = Seq(10),
+          // we read the checkpoint twice: once for the P &M and once for the scan files
+          expParquetReadSetSizes = Seq(1, 1),
+          expLastCheckpointReadCalls = Some(1)
```

Comment:
worth a simple comment here:
- if the file does not exist, we try reading it once, realize it doesn't exist, and don't try reading it again
- if the file does exist, we succeed reading it the first time